src/main/scala/sbtassembly/Shader.scala

```diff
@@ -81,7 +81,14 @@ private[sbtassembly] object Shader {
     }}
 
     val proc = JJProcessor(jjrules, verbose = level == Level.Debug, true)
-    val files = AssemblyUtils.getMappings(dir, Set())
+
+    /*
+    jarjar MisplacedClassProcessor class transforms byte[] to a class using org.objectweb.asm.ClassReader.getClassName
+    which always translates class names containing '.' into '/', regardless of OS platform.
+    We need to transform any windows file paths in order for jarjar to match them properly and not omit them.
+     */
+    val files = AssemblyUtils.getMappings(dir, Set()).map(f =>
+      if (f._2.contains('\\')) (f._1, f._2.replace('\\', '/')) else f)
```

Comment:
Ok, During investigation of the issue, I see that shading rules only execute if this `if` clause falls through:

// If the jar name path does not exist, or is not for this jar, unzip the jar
`if (!ao.cacheUnzip || !jarNamePath.exists || IO.read(jarNamePath) != jar.data.getCanonicalPath)`

Otherwise, it won't fall through. This means that if I have the JAR i'm trying to Shade on the target assemblyOption, it won't shade the directory and put it inside the JAR. I'm not sure if this is expected behavior, it seems weird. Should there be an added check if the said JAR is on the shadeRules list?